backend-lib/src/main/rust/lib.rs

```diff
@@ -1714,6 +1714,49 @@ fn zip317_helper<DbT>(
     )
 }
```

Comment:
This is now always called with `use_zip317_fees == JNI_TRUE`, and is private. Consider:
```rust
fn zip317_helper<DbT>(
    change_memo: Option<MemoBytes>,
) -> GreedyInputSelector<DbT, SingleOutputChangeStrategy> {
    GreedyInputSelector::new(
        SingleOutputChangeStrategy::new(StandardFeeRule::Zip317, change_memo, ShieldedProtocol::Orchard),
        DustOutputPolicy::default(),
    )
}
```
Also, `zip317_input_selector` would be a better name.